tests/functional/test_install_vcs_git.py

```diff
@@ -80,15 +97,9 @@ def test_install_editable_from_git_with_https(script, tmpdir):
     """
     Test cloning from Git with https.
     """
-    result = script.pip(
-        'install', '-e',
-        '%s#egg=pip-test-package' %
-        local_checkout(
-            'git+https://github.com/pypa/pip-test-package.git',
-            tmpdir.join("cache"),
-        ),
-        expect_error=True,
-    )
+    url = 'git+https://github.com/pypa/pip-test-package.git'
+    local_url = _checkout_into_temp(url, tmpdir, egg='pip-test-package')
```

Comment:
Since all these URLs are coming from Github, perhaps we can just put that into the helper as well?

Reply:
Sure. I'll see what I can do. And I think over time the idea would be to see if some of the tests can avoid hitting GitHub altogether.